mpf/devices/multiball_lock.py

```diff
@@ -305,3 +306,12 @@ def _post_events(self, device, **kwargs):
     def _request_new_balls(self, balls):
         """Request new ball to playfield."""
         self.source_playfield.add_ball(balls=balls)
+    
+    def compensate_lock(self):
+        # evilmopey solution for issue 1278
+        if self.config['locked_ball_counting_strategy'] == "virtual_only" or "min_virtual_physical":
+            if self._locked_balls > self._physically_locked_balls:
+                # This means we need to add more balls to the physical lock
+                ball_deficit = self._locked_balls - self._physically_locked_balls
+                BallDevice.request_ball(self.lock_devices[0], ball_deficit)
```

Comment:
How does `request_ball` work for locks that don't have a route from the trough? Will it be ignored, or wait until a playfield ball lands in it?

Reply:
So when request_ball is called it starts a for loop that iterates as many times as the value balls were passed in. Each iteration calls setup_queue_eject_to_target, if the ball cannot be routed to the target a tuple entry will then be made into the _ball_requests_ deque. So to my knowledge the request is not being ignored.